### PR TITLE
storage: add source/sink utilization views

### LINKED ISSUE #15808: [Epic] Expose CPU and memory usage of clusters and storage hosts in system catalog
### Product Outcome

We will expose the CPU and memory usage of clusters and storage hosts in the system catalog so users can monitor their usage.

### Specification

Subset of https://github.com/MaterializeInc/cloud/issues/4120

### Design

Subset of https://github.com/MaterializeInc/cloud/issues/4120

### Documentation

- [x] Covered in https://github.com/MaterializeInc/materialize/pull/16209 for compute
- [ ] Storage hosts outstanding

### Required Issues (Must Haves)
- [x] https://github.com/MaterializeInc/materialize/issues/16323 (for storage)
- [x] https://github.com/MaterializeInc/materialize/issues/16477
- [x] Improve size mapping table to show all available sizes (pending PR)
- [ ] Add `mz_source_utilization` and `mz_sink_utilization` convenience views (based on `mz_storage_host_metrics`)

### Blockers

None

### PR DESCRIPTION
**Pull request by aljoscha**
### Motivation

Closes #15808 (by adding the last outstanding bits)

### Tips for reviewer

I added a static `mz_internal.mz_storage_host_sizes`, similar to how @umanwizard did it for cluster replicas, and then added the convenience views that use it and `mz_internal.mz_storage_host_metrics`.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  -

### CODE REVIEW

**Comment by aljoscha** [comment]
Not exactly sure who owns these views/tables, but I tagged @jkosh44. Please let me know if I should ping someone else. 😅 

cc @ggnall, once this is in, UX should be able to build shweet dashboards

**Review by jkosh44 — APPROVED** [review]
LGTM, can you add a test that asserts the column values are non-zero.

**Comment by aljoscha** [comment]
> LGTM, can you add a test that asserts the column values are non-zero.

ty! where would we want that test to live?

**Comment by jkosh44** [comment]
> > LGTM, can you add a test that asserts the column values are non-zero.
> 
> ty! where would we want that test to live?

I think you can probably put it in the same place @umanwizard  put the tests for replicas, `metrics.td`. See https://github.com/MaterializeInc/materialize/pull/16667/files for details.

File: test/testdrive/metrics-storage.td [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+> CREATE SINK metrics_test_sink FROM metrics_test_source
+  INTO KAFKA CONNECTION kafka_conn (TOPIC 'testdrive-metrics-test-sink-${testdrive.seed}')
+  FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY CONNECTION csr_conn
+  ENVELOPE DEBEZIUM
+
+# NOTE: These queries are slow to succeed because the default metrics scraping
+# interval is 20 seconds.
+
+> SELECT s.name, u.cpu_percent >= 0.0, u.memory_percent > 0.0
```
Comment (umanwizard) [inline test/testdrive/metrics-storage.td:35]: Why `>=` and not `>` ?

Comment (aljoscha) [reply test/testdrive/metrics-storage.td:35]: talked about this offline: storage hosts sit idle more often than compute replicas so their CPU usage _will_ often be `0.0`.

**Review by umanwizard — APPROVED** [review]
